libs/vkd3d/device_vkd3d_ext.c

```diff
@@ -434,7 +446,61 @@ static HRESULT STDMETHODCALLTYPE d3d12_dxvk_interop_device_UnlockCommandQueue(d3
     return S_OK;
 }
 
-CONST_VTBL struct ID3D12DXVKInteropDeviceVtbl d3d12_dxvk_interop_device_vtbl =
+static HRESULT STDMETHODCALLTYPE d3d12_dxvk_interop_device_CreateInteropCommandQueue(d3d12_dxvk_interop_device_iface *iface,
+        const D3D12_COMMAND_QUEUE_DESC *desc, uint32_t vk_family_index, ID3D12CommandQueue **command_queue)
+{
+    struct d3d12_device *device = d3d12_device_from_ID3D12DXVKInteropDevice(iface);
+    struct d3d12_command_queue *object;
+    HRESULT hr;
+
+    TRACE("iface %p, desc %p, vk_family %d, command_queue %p.\n",
+            iface, desc, command_queue);
+
+    hr = d3d12_command_queue_create(device, desc, vk_family_index, &object);
+    if (FAILED(hr))
+        return hr;
+
+    *command_queue = (ID3D12CommandQueue *)&object->ID3D12CommandQueue_iface;
+
+    return S_OK;
+}
+
+static HRESULT STDMETHODCALLTYPE d3d12_dxvk_interop_device_CreateInteropCommandAllocator(d3d12_dxvk_interop_device_iface *iface,
+        D3D12_COMMAND_LIST_TYPE type, uint32_t vk_family_index, ID3D12CommandAllocator **command_allocator)
+{
+    struct d3d12_device *device = d3d12_device_from_ID3D12DXVKInteropDevice(iface);
+    struct d3d12_command_allocator *object;
+    HRESULT hr;
+
+    TRACE("iface %p, type %x, vk_family %d, command_allocator %p.\n",
+            iface, type, command_allocator);
+
+    hr = d3d12_command_allocator_create(device, type, vk_family_index, &object);
+    if (FAILED(hr))
+        return hr;
+
+    *command_allocator = (ID3D12CommandAllocator *)&object->ID3D12CommandAllocator_iface;
+
+    return S_OK;
+}
+
+static HRESULT STDMETHODCALLTYPE d3d12_dxvk_interop_device_BeginVkCommandBufferInterop(d3d12_dxvk_interop_device_iface *iface,
+        ID3D12CommandList *pCmdList, VkCommandBuffer *cmdBuf)
+{
+    struct d3d12_command_list *cmd_list;
+    cmd_list = d3d12_command_list_from_iface(pCmdList);
+
+    if (!cmd_list)
+        return E_INVALIDARG;
+
+    d3d12_command_list_invalidate_all_state(cmd_list);
```

Comment:
This isn't quite enough to make the interop work, but it's a start. We can probably flesh that out after this lands. Just add a FIXME comment here for now.